Scripts/Runtime/Logging/.CSProject/UnityLogListener.cs

```diff
@@ -0,0 +1,99 @@
+    {
+        private static UnityLogListener s_Instance;
+
+        // With the Burst compiler, it's common to disable the Domain Reload step when entering play mode to increase iteration time.
+        // https://docs.unity3d.com/Packages/com.unity.entities@0.17/manual/install_setup.html
+        // Because of this, statics will preserve state across play sessions.
+        // `RuntimeInitializeOnLoadMethod` ensures we reset the state on every play session.
+        // `RuntimeInitializeLoadType.SubsystemRegistration` represents the earliest moment of all the entries in the enum.
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
+        private static void Init()
+        {
+            s_Instance = null;
+        }
+
+        // The log handler in place before this listener initialized.
+        // Called when a log comes through that this listener has already handled.
+        private readonly UnityEngine.ILogHandler m_ExistingLogHandler;
+
+        public UnityLogListener()
+        {
+            Debug.Assert(s_Instance == null, $"There can only be one instance of the {nameof(UnityLogListener)} and it has already been initialized.");
+            s_Instance = this;
+
+            m_ExistingLogHandler = Debug.unityLogger.logHandler;
+            Debug.unityLogger.logHandler = this;
+        }
+
+        /// <inheritdoc />
+        public void LogFormat(LogType logType, UnityEngine.Object context, string format, params object[] args)
+        {
+            // Bypass redirecting unity log messages to logger if logger is already handling the log.
+            // This happens when UnityLogHandler is being used.
+            if (Log.IsHandlingLog)
+            {
+                m_ExistingLogHandler.LogFormat(logType, context, format, args);
+                return;
+            }
+
+            StackFrame stackFrame = GetLogCallerStackFrame();
+            object loggerContext = ResolveLogContext(context, stackFrame);
+            Log.GetLogger(loggerContext).AtLevel(LogTypeToLogLevel(logType), string.Format(format, args), stackFrame.GetFileName(), stackFrame.GetMethod().Name, stackFrame.GetFileLineNumber());
+        }
+
+        /// <inheritdoc />
+        public void LogException(Exception exception, UnityEngine.Object context)
+        {
+            // Bypass redirecting unity log messages to logger if logger is already handling the log.
+            // This happens when UnityLogHandler is being used.
+            if (Log.IsHandlingLog)
+            {
+                m_ExistingLogHandler.LogException(exception, context);
+                return;
+            }
+
+            StackFrame stackFrame = GetLogCallerStackFrame();
+            object loggerContext = ResolveLogContext(context, stackFrame);
+            Log.GetLogger(loggerContext).Error(exception.ToString(), stackFrame.GetFileName(), stackFrame.GetMethod().Name, stackFrame.GetFileLineNumber());
```

Comment:
I feel like these three lines and lines 48 to 50 could be consolidated into one function that both `LogFormat` and `LogException` call?

Reply:
I like that idea. 
The `GetLogCallerStackFrame` number gets even more ambiguous/"magic" then but I think if I pass in the depth from here that'll be fine.

Reply:
Fixed. 
Forced me to figure out what exactly was happening at frame 2 :)
Also decided against the passed in skipFrames parameter. The comments are sufficient for now.